src/Microsoft.CodeAnalysis.Razor/ComponentTagHelperDescriptorProvider.cs

```diff
@@ -0,0 +1,132 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Razor.Language;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    public class ComponentTagHelperDescriptorProvider : RazorEngineFeatureBase, ITagHelperDescriptorProvider
```

Comment:
Haven't added tests for this. It needs a lot of work.

Reply:
I think it's fine to copy-paste these Blazor compiler bits without tests for now. We'll do the work to reconcile this later.

Reply:
Is there a **SUPER** good reason to have this public?

Reply:
It's used by `GenerateCommand` which does not have IVT to this assembly.

Reply:
Tracking it here - https://github.com/aspnet/Razor/issues/2744